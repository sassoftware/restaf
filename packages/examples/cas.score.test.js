/*
 * ------------------------------------------------------------------------------------
 *   Copyright Â© 2023, SAS Institute Inc., Cary, NC, USA. *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 * ---------------------------------------------------------------------------------------
 *
 */

'use strict';
let testFunctions = require('./testFunctions');
let testInfo;
beforeAll(async () => {
	testInfo = await require('./lib/setupAll')();
});

test('Upload sasast and score', async () => {
	expect.assertions();
	let l = await testFunctions.casScoreAst(false, testInfo);
	expect(l).toMatchSnapshot();
});


test('Upload hdat and score', async () => {
	expect.assertions();
	let l = await testFunctions.casScoreHdat(false, testInfo);
	expect(l).toMatchSnapshot();
});