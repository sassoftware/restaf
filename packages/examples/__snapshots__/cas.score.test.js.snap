// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload hdat and score 1`] = `
Object {
  "P_BAD": 0.30567335603563,
  "_Index_": 1,
  "_WARN_": "",
}
`;

exports[`Upload sasast and score 1`] = `
Object {
  "describe": Array [
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "type_n        ",
      "RawType": "Num",
      "Role": "Input",
      "Type": "Interval      ",
    },
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "amount        ",
      "RawType": "Num",
      "Role": "Input",
      "Type": "Interval      ",
    },
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "newbalanceDest",
      "RawType": "Num",
      "Role": "Input",
      "Type": "Interval      ",
    },
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "newbalanceOrig",
      "RawType": "Num",
      "Role": "Input",
      "Type": "Interval      ",
    },
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "oldbalanceDest",
      "RawType": "Num",
      "Role": "Input",
      "Type": "Interval      ",
    },
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "oldbalanceOrg ",
      "RawType": "Num",
      "Role": "Input",
      "Type": "Interval      ",
    },
    Object {
      "FormatName": "         ",
      "Length": 8,
      "Name": "isFraud       ",
      "RawType": "Num",
      "Role": "Id   ",
      "Type": "              ",
    },
  ],
  "score": Object {
    "_Index_": 1,
    "_SVDDDISTANCE_": 0.97031525978426,
    "_SVDDSCORE_": -1,
    "amount": 0,
    "isFraud": 0,
    "newbalanceDest": 0,
    "newbalanceOrig": 0,
    "oldbalanceDest": 0,
    "oldbalanceOrg": 0,
    "type_n": 0,
  },
}
`;
